File: app/src/main/kotlin/net/kibotu/android/recyclerviewpresenter/app/java/PhotoPresenter.java
package net.kibotu.android.recyclerviewpresenter.app.java;

import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import net.kibotu.android.recyclerviewpresenter.Presenter;
import net.kibotu.android.recyclerviewpresenter.PresenterViewModel;
import net.kibotu.android.recyclerviewpresenter.app.R;
import net.kibotu.android.recyclerviewpresenter.app.misc.GlideApp;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import static com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions.withCrossFade;

/**
 * Created by <a href="https://kibotu.net">Jan Rabe</a>.
 */

public class PhotoPresenter extends Presenter<String> {

    public PhotoPresenter() {
        super(R.layout.photo_presenter_item);
    }

    @Override
    public void bindViewHolder(@NotNull RecyclerView.ViewHolder viewHolder, @NotNull PresenterViewModel<String> item, @Nullable List<Object> payloads) {
        ImageView photo = viewHolder.itemView.findViewById(R.id.photo);

        GlideApp.with(viewHolder.itemView.getContext().getApplicationContext())
                .load(item.getModel())
                .transition(withCrossFade())
                .into(photo)
                .waitForLayout()
                .clearOnDetach();
    }
}

File: app/src/main/java/net/kibotu/android/recyclerviewpresenter/app/kotlin/PhotoPresenter.kt
package net.kibotu.android.recyclerviewpresenter.app.kotlin

import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import kotlinx.android.synthetic.main.photo_presenter_item.view.*
import net.kibotu.android.recyclerviewpresenter.Adapter
import net.kibotu.android.recyclerviewpresenter.Presenter
import net.kibotu.android.recyclerviewpresenter.PresenterModel
import net.kibotu.android.recyclerviewpresenter.app.R
import net.kibotu.android.recyclerviewpresenter.app.misc.GlideApp

/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */
class PhotoPresenter : Presenter<String>() {

    override val layout: Int = R.layout.photo_presenter_item

    override fun bindViewHolder(viewHolder: RecyclerView.ViewHolder, item: PresenterModel<String>, position: Int, payloads: MutableList<Any>?, adapter: Adapter): Unit = with(viewHolder.itemView) {
        GlideApp.with(this.context.applicationContext)
            .load(item.model)
            .transition(DrawableTransitionOptions.withCrossFade())
            .into(photo)
    }
}